### ISSUE TITLE
create-shape function is missing

### ISSUE THREAD

**Issue by blueberry** [quil/quil#165]
There is currently no equivalent to Processing's createShape method. Is it intentional, or simply an oversight?

**Comment by blueberry** [comment]
Now I see that other methods of PShape are also missing in quil: endShape, setStroke, setFill etc...

**Comment by blueberry** [comment]
And another: vertex in quil is hardcoded to use (current-graphics), while in Processing it can also work on a PShape. That use case is important for a little bit cleaner (although also non-pure) functional style...

**Comment by nbeloglazov** [comment]
Yes, I believe the main reason behind not adding `createShape` is that Quil doesn't have API for working with PShape object. Though you still can access them via java interop:
```clojure
(let [shape (.createShape (q/current-graphics))]
  (doto shape
    (.beginShape)
    (.noStroke)     
    ...))
```

So working with shapes is totally possible but in not a very clojury way. Pull requests with adding API (may via new namespace) for working with shapes is welcome! 

As for `vertex`, yes it was designed to work with current-graphics (either applet instance or custom graphics object), but I think you can also use it with shape:
```clojure
(q/with-graphics my-shape
  (q/vertex 1 2 3)
  ...))
```

**Comment by blueberry** [comment]
Thank you.
Is there any particular reason for not providing PShape support?
I'll probably first experiment with PShape Java interop, and maybe create a mini clojure wrapper, and if/when that show value, I'll be glad if it is merged into quil. Even if it stays separate, I will try to make it compatible, so they can be used side to side.

**Comment by nbeloglazov** [comment]
I think the reasons was that no one thought it was important enough to work on it. So you can say it is laziness :)
